interop-testing/src/main/java/io/grpc/testing/integration/Http2Client.java

```diff
@@ -0,0 +1,336 @@
+    }
+    if (usage) {
+      Http2Client c = new Http2Client();
+      System.out.println(
+          "Usage: [ARGS...]"
+              + "\n"
+              + "\n  --server_host=HOST          Server to connect to. Default " + c.serverHost
+              + "\n  --server_port=PORT          Port to connect to. Default " + c.serverPort
+              + "\n  --test_case=TESTCASE        Test case to run. Default " + c.testCase
+              + "\n    Valid options:"
+              + validTestCasesHelpText()
+      );
+      System.exit(1);
+    }
+  }
+
+  private void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+    asyncStub = TestServiceGrpc.newStub(channel);
+  }
+
+  private synchronized void tearDown() {
+    try {
+      if (channel != null) {
+        channel.shutdown();
+      }
+    } catch (RuntimeException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private void run() {
+    System.out.println("Running test " + testCase);
+    try {
+      runTest(Http2TestCases.fromString(testCase));
+    } catch (RuntimeException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+    System.out.println("Test completed.");
+  }
+
+  private void runTest(Http2TestCases testCase) throws Exception {
+    switch (testCase) {
+      case RST_AFTER_HEADER:
+        tester.rstStreamAfterHeader();
+        break;
+      case RST_AFTER_DATA:
+        tester.rstStreamAfterData();
+        break;
+      case RST_DURING_DATA:
+        tester.rstStreamDuringData();
+        break;
+      case GOAWAY:
+        tester.goAway();
+        break;
+      case PING:
+        tester.ping();
+        break;
+      case MAX_STREAMS:
+        tester.maxStreams();
+        break;
+      default:
+        throw new IllegalArgumentException("Unknown test case: " + testCase);
+    }
+  }
+
+  private class Tester {
+    private final SimpleRequest simpleRequest = SimpleRequest.newBuilder()
+        .setResponseSize(314159)
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .setPayload(Payload.newBuilder()
+            .setBody(ByteString.copyFrom(new byte[271828])))
+        .build();
+
+    public void rstStreamAfterHeader() throws Exception {
+      try {
+        blockingStub.unaryCall(simpleRequest);
+        fail("Expected call to fail");
+      } catch (StatusRuntimeException ex) {
+        assertEquals(Status.Code.UNAVAILABLE, ex.getStatus().getCode());
+      }
+    }
+
+    public void rstStreamAfterData() throws Exception {
+      try {
+        blockingStub.unaryCall(simpleRequest);
+        fail("Expected call to fail");
+      } catch (StatusRuntimeException ex) {
+        assertEquals(Status.Code.UNAVAILABLE, ex.getStatus().getCode());
+      }
+
+      // Use async stub to verify data was received.
+      StreamRecorder<SimpleResponse> responseObserver = StreamRecorder.create();
```

Comment:
@ejona86  declined PR #2385 on the grounds that StreamRecoder was going to be deleted.  

You probably don't want to use it here.